filebeat/harvester/reader/multiline.go

```diff
@@ -114,31 +114,34 @@ func (mlr *Multiline) Next() (Message, error) {
 
 func (mlr *Multiline) readFirst() (Message, error) {
 	for {
-		p, err := mlr.reader.Next()
-		if err == nil {
-			if p.Bytes == 0 {
+		message, err := mlr.reader.Next()
+		if err != nil {
+			// no lines buffered -> ignore timeout
+			// TODO: Does this only happen on backoff? Normally is EOF?
```

Comment:
we do not care about backoff or any behavior of underlying reader at all in multiline-reader. `errMultilineTimeout` is always triggered if no line and no error is returned by underlying reader for a duration of `timeout`. The errMultilineTimeout is internal to multiline reader in order to distinguish the timer triggering from actual errors. So to say, errMultilineTimeout is no error(!), but a signal saying: no return from underlying reader for N seconds.